zerver/webhooks/freshstatus/view.py

```diff
@@ -0,0 +1,164 @@
+            services_number=len(services_data) - FRESHSTATUS_SERVICES_LIMIT,
+        )
+    return services_content.rstrip()
+
+
+def get_subject_for_http_request(payload: Dict[str, Any]) -> str:
+    event_data = payload["event_data"]
+    if event_data["event_type"] == "INCIDENT_OPEN" and payload["id"] == "1":
+        subject = FRESHSTATUS_TOPIC_TEMPLATE_TEST
+    else:
+        subject = FRESHSTATUS_TOPIC_TEMPLATE.format(title=payload["title"])
+
+    return subject
+
+
+def get_body_for_http_request(payload: Dict[str, Any]) -> str:
+    event_data = payload["event_data"]
+    if event_data["event_type"] == "INCIDENT_OPEN" and payload["id"] == "1":
+        body = FRESHSTATUS_MESSAGE_TEMPLATE_TEST
+    else:
+
+        if event_data["event_type"] == "INCIDENT_OPEN":
+            services_data = []
+            for service in payload["affected_services"].split(","):
+                services_data.append({"service_name": service})
+            data = {
+                "title": payload["title"],
+                "description": payload["description"],
+                "start_time": dateutil.parser.parse(payload["start_time"]).strftime(
+                    "%d/%m/%Y %H:%M %Z"
+                ),
+                "affected_services": get_services_content(services_data),
+            }
+            body = FRESHSTATUS_MESSAGE_TEMPLATE_INCIDENT_OPEN.format(**data)
+        elif event_data["event_type"] == "INCIDENT_NOTE_CREATE":
+            if payload["incident_status"] == "Closed":
+                body = FRESHSTATUS_MESSAGE_TEMPLATE_INCIDENT_CLOSED.format(**payload)
+            elif payload["incident_status"] == "Open":
+                body = FRESHSTATUS_MESSAGE_TEMPLATE_INCIDENT_NOTE_CREATED.format(**payload)
+        elif event_data["event_type"] == "MAINTENANCE_PLANNED":
+            services_data = []
+            for service in payload["affected_services"].split(","):
+                services_data.append({"service_name": service})
+            data = {
+                "title": payload["title"],
+                "description": payload["description"],
+                "scheduled_start_time": dateutil.parser.parse(
+                    payload["scheduled_start_time"]
+                ).strftime("%d/%m/%Y %H:%M %Z"),
+                "scheduled_end_time": dateutil.parser.parse(payload["scheduled_end_time"]).strftime(
+                    "%d/%m/%Y %H:%M %Z"
+                ),
+                "affected_services": get_services_content(services_data),
+            }
+            body = FRESHSTATUS_MESSAGE_TEMPLATE_SCHEDULED_MAINTENANCE_PLANNED.format(**data)
+        elif event_data["event_type"] == "MAINTENANCE_NOTE_CREATE":
+            if payload["incident_status"] == "Closed":
+                body = FRESHSTATUS_MESSAGE_TEMPLATE_SCHEDULED_MAINTENANCE_CLOSED.format(**payload)
+            else:
+                body = FRESHSTATUS_MESSAGE_TEMPLATE_SCHEDULED_MAINTENANCE_NOTE_CREATED.format(
+                    **payload
+                )
```

Comment:
Is it intentional that this just falls through for other event types?

Reply:
Yes, this is intentional. Under the `event_type` of `MAINTENANCE_NOTE_CREATE` it can either be a note which closes or its just a note, and its under the `elif` so its not a catch of all any event, just of a note that isn't closed.

Reply:
Maybe worth a comment about that intent.